addons/LPCAnimatedSprite/LPCSpriteSheet.gd

```diff
@@ -47,47 +52,110 @@ var NormalAnimationData:Array[LPCAnimationData] = [
+	LPCAnimationData.new(6,"SLASH_LEFT",22, 0,false, 192),
+	LPCAnimationData.new(6,"SLASH_DOWN",23, 0,false, 192),
+	LPCAnimationData.new(6,"SLASH_RIGHT",24, 0,false, 192),
+]
+
+static var PrecompiledOversizedSlashReverse:Array[LPCAnimationData] = [
+	LPCAnimationData.new(6,"SLASH_REVERSE_UP",25, 0,false, 192),
+	LPCAnimationData.new(6,"SLASH_REVERSE_LEFT",26, 0,false, 192),
+	LPCAnimationData.new(6,"SLASH_REVERSE_DOWN",27, 0,false, 192),
+	LPCAnimationData.new(6,"SLASH_REVERSE_RIGHT",28, 0,false, 192),
+]
+
+static var PrecompiledOversizedThrust:Array[LPCAnimationData] = [
+	LPCAnimationData.new(6,"THRUST_UP",29, 0,false, 192),
+	LPCAnimationData.new(6,"THRUST_LEFT",30, 0,false, 192),
+	LPCAnimationData.new(6,"THRUST_DOWN",31, 0,false, 192),
+	LPCAnimationData.new(6,"THRUST_RIGHT",32, 0,false, 192),
 ]
-var ThrustAnimationData:Array[LPCAnimationData] = [
+
+static var ThrustAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(6,"THRUST_UP",0, 0,false),
 	LPCAnimationData.new(6,"THRUST_LEFT",1, 0,false),
 	LPCAnimationData.new(6,"THRUST_DOWN",2, 0,false),
 	LPCAnimationData.new(6,"THRUST_RIGHT",3, 0,false),
 ]
-var RodAnimationData:Array[LPCAnimationData] = [
+static var RodAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(6,"ROD_UP",0, 0,false),
 	LPCAnimationData.new(6,"ROD_LEFT",1, 0,false),
 	LPCAnimationData.new(6,"ROD_DOWN",2, 0,false),
 	LPCAnimationData.new(6,"ROD_RIGHT",3, 0,false),
 ]
-var WhipAnimationData:Array[LPCAnimationData] = [
+static var WhipAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(8,"WHIP_UP",0, 0,false),
 	LPCAnimationData.new(8,"WHIP_LEFT",1, 0,false),
 	LPCAnimationData.new(8,"WHIP_DOWN",2, 0,false),
 	LPCAnimationData.new(8,"WHIP_RIGHT",3, 0,false),
 ]
 
-func Size() -> int:
-	match SpriteType:
-		SpriteTypeEnum.Normal:
-			return 64
-		_:
-			return 192
+enum SpritesheetType
+{
+	SizeNormal,
+	Size_6_6_8
+}
+
+func GetSpritesheetType():
+	if SpriteSheet.get_height() == 3648:
+		return SpritesheetType.Size_6_6_8
+	return SpritesheetType.SizeNormal
+	
+func GetSpriteRect(animationData:LPCAnimationData, frame: int) -> Rect2:
+	const startOfOversizedAnimation = 21;
+	var spriteSize:int = animationData.Size
+	match GetSpritesheetType():
+		LPCSpriteSheet.SpritesheetType.SizeNormal:
+			return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
+		LPCSpriteSheet.SpritesheetType.Size_6_6_8:
+			if animationData.Row < startOfOversizedAnimation:
+				return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
+			else:
+				var startPosition:int = startOfOversizedAnimation * 64
+				var row = animationData.Row - startOfOversizedAnimation
+				return Rect2(spriteSize*(frame+animationData.Col), startPosition + spriteSize*row, spriteSize, spriteSize)
+	return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
```

Comment:
This exact line is repeated twice before (+ the differing version), do you foresee needing the space for further logic or could the code be simplified to have only one copy of this call at the end (i.e. if type == _6_6_8 AND .row<startofOversized)